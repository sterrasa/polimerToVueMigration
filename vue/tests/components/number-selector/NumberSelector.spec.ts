import { describe, it, beforeEach, afterEach, vi } from 'vitest';
import { expect } from 'chai'; // Import assertion library
import { mount } from '@vue/test-utils'

import NumberSelector from '@/components/number-selector/NumberSelector.ce.vue';

describe('NumberSelector Test suite', () => {

    let consoleWarnSpy;

    beforeEach(() => {
        consoleWarnSpy = vi.spyOn(console, 'warn');
    });

    afterEach(() => {
        consoleWarnSpy.mockRestore();
    });

    it('should render the webcomponent without errors', () => {
        const wrapper = mount(NumberSelector, {
            props: {
                minValue: 0,
                maxValue: 10,
            },
        });
        expect(wrapper.element).toMatchSnapshot()
    });

    it('should render MainSelector with the correct length', () => {
        const minValue = 0;
        const maxValue = 10;
        const wrapper = mount(NumberSelector, {
            props: {
                minValue,
                maxValue,
            },
        });
        const selectables = wrapper.element.querySelectorAll('.selectable');
        expect(selectables).to.have.lengthOf(maxValue - minValue + 1);
    });

    it('should render the correct slot message', () => {
        const message = 'Please select a number!';
        const wrapper = mount(NumberSelector, {
            slots: {
                default: message,
            },
        });
        const slotContent = wrapper.find('.number-selector').text();
        console.log(slotContent)
        expect(slotContent.trim()).to.equal(message);

    });

    it('should emit the "selected" event with the correct number when a number is selected', async () => {
        const wrapper = mount(NumberSelector, {
            props: {
                minValue: 1,
                maxValue: 10,
            },
        });

        const inputElement = wrapper.findAll('input[type="radio"]').at(2);
        await inputElement.trigger('change');

        expect(wrapper.emitted()).toHaveProperty('selected');
        expect(wrapper.emitted('selected')).toHaveLength(1);

        const emittedValue = wrapper.emitted('selected')[0][0];
        expect(typeof emittedValue).toBe('number');
        expect(emittedValue).toBe(3);
    });

    it('should log a warning when minValue or maxValue prop is missing', () => {
        mount(NumberSelector);
        const warningMessageMinValue = '[Vue warn]: Missing required prop: "minValue"';
        const warningMessageMaxValue = '[Vue warn]: Missing required prop: "maxValue"';
        const hasMinValueWarning = consoleWarnSpy.mock.calls.some(call => call[0].includes(warningMessageMinValue));
        const hasMaxValueWarning = consoleWarnSpy.mock.calls.some(call => call[0].includes(warningMessageMaxValue));

        expect(hasMinValueWarning && hasMaxValueWarning).to.be.true;

    });

});
