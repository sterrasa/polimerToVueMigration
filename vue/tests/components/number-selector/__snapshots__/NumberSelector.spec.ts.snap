// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NumberSelector Test suite > should render the webcomponent without errors 1`] = `
<div
  class="number-selector"
  data-v-5fddf4a8=""
  style="--background-color: #fffff; --text-color: rgba(0 0 0 / 87%);"
>
  
  Select:
  
  <div
    class="selectables"
    data-v-5fddf4a8=""
  >
    
    <span
      class="selectable"
      data-v-5fddf4a8=""
    >
      <input
        aria-label="Select 0"
        data-v-5fddf4a8=""
        id="item-0"
        style="width: 1.0rem; height: 1.0rem;"
        type="radio"
        value="0"
      />
      <label
        data-v-5fddf4a8=""
        for="item-0"
      >
        0
      </label>
    </span>
    <span
      class="selectable"
      data-v-5fddf4a8=""
    >
      <input
        aria-label="Select 1"
        data-v-5fddf4a8=""
        id="item-1"
        style="width: 1.0rem; height: 1.0rem;"
        type="radio"
        value="1"
      />
      <label
        data-v-5fddf4a8=""
        for="item-1"
      >
        1
      </label>
    </span>
    <span
      class="selectable"
      data-v-5fddf4a8=""
    >
      <input
        aria-label="Select 2"
        data-v-5fddf4a8=""
        id="item-2"
        style="width: 1.0rem; height: 1.0rem;"
        type="radio"
        value="2"
      />
      <label
        data-v-5fddf4a8=""
        for="item-2"
      >
        2
      </label>
    </span>
    <span
      class="selectable"
      data-v-5fddf4a8=""
    >
      <input
        aria-label="Select 3"
        data-v-5fddf4a8=""
        id="item-3"
        style="width: 1.0rem; height: 1.0rem;"
        type="radio"
        value="3"
      />
      <label
        data-v-5fddf4a8=""
        for="item-3"
      >
        3
      </label>
    </span>
    <span
      class="selectable"
      data-v-5fddf4a8=""
    >
      <input
        aria-label="Select 4"
        data-v-5fddf4a8=""
        id="item-4"
        style="width: 1.0rem; height: 1.0rem;"
        type="radio"
        value="4"
      />
      <label
        data-v-5fddf4a8=""
        for="item-4"
      >
        4
      </label>
    </span>
    <span
      class="selectable"
      data-v-5fddf4a8=""
    >
      <input
        aria-label="Select 5"
        data-v-5fddf4a8=""
        id="item-5"
        style="width: 1.0rem; height: 1.0rem;"
        type="radio"
        value="5"
      />
      <label
        data-v-5fddf4a8=""
        for="item-5"
      >
        5
      </label>
    </span>
    <span
      class="selectable"
      data-v-5fddf4a8=""
    >
      <input
        aria-label="Select 6"
        data-v-5fddf4a8=""
        id="item-6"
        style="width: 1.0rem; height: 1.0rem;"
        type="radio"
        value="6"
      />
      <label
        data-v-5fddf4a8=""
        for="item-6"
      >
        6
      </label>
    </span>
    <span
      class="selectable"
      data-v-5fddf4a8=""
    >
      <input
        aria-label="Select 7"
        data-v-5fddf4a8=""
        id="item-7"
        style="width: 1.0rem; height: 1.0rem;"
        type="radio"
        value="7"
      />
      <label
        data-v-5fddf4a8=""
        for="item-7"
      >
        7
      </label>
    </span>
    <span
      class="selectable"
      data-v-5fddf4a8=""
    >
      <input
        aria-label="Select 8"
        data-v-5fddf4a8=""
        id="item-8"
        style="width: 1.0rem; height: 1.0rem;"
        type="radio"
        value="8"
      />
      <label
        data-v-5fddf4a8=""
        for="item-8"
      >
        8
      </label>
    </span>
    <span
      class="selectable"
      data-v-5fddf4a8=""
    >
      <input
        aria-label="Select 9"
        data-v-5fddf4a8=""
        id="item-9"
        style="width: 1.0rem; height: 1.0rem;"
        type="radio"
        value="9"
      />
      <label
        data-v-5fddf4a8=""
        for="item-9"
      >
        9
      </label>
    </span>
    <span
      class="selectable"
      data-v-5fddf4a8=""
    >
      <input
        aria-label="Select 10"
        data-v-5fddf4a8=""
        id="item-10"
        style="width: 1.0rem; height: 1.0rem;"
        type="radio"
        value="10"
      />
      <label
        data-v-5fddf4a8=""
        for="item-10"
      >
        10
      </label>
    </span>
    
  </div>
</div>
`;
